docs/develop/howtos/route_migration.md

```diff
@@ -0,0 +1,63 @@
+# Migrate legacy routes
+
+This guide describes the process to migrate legacy routes using the redirector module.
+
+## Redirector module
+
+The redirector module registers legacy routes (called **source**) and the redirection to a new **target**.
+
+Redirection rules are retrieved by the application from the config `REDIRECTOR_RULES`.
+
+## Steps
+
+- Identify the old endpoint url (e.g. `/old_endpoint`)
+- Identify the new endpoint url (e.g. where to redirect a request to `/old_endpoint`).
+- Add the config `REDIRECTOR_RULES` in the application (e.g. `my_syte/config.py`). E.g:
+  ```python
+    REDIRECTOR_RULES = {
+        "redirect_my_rule": {
+            "source": "/old_endpoint",
+            "target": "/new_endpoint"
+        }
+    }
+  ```
+
+## Examples
+
+Sometimes it is necessary to modify the request further (e.g. map old parameters to newer ones). The redirector module also accepts a function that returns an URL as the rule's `target`.
+Find below some examples
+
+- Redirect a request to a new endpoint.
```

Comment:
```suggestion
- Redirect a request to a new endpoint keeping the search arguments.
```

Isn't this the default behaviour when we define source and target as strings? Or we just redirect to the string without any parameter passed?

Reply:
We just redirect a `source` to a `target`. 
See: https://github.com/inveniosoftware/invenio-app-rdm/blob/1425f0049e375c5dc428e471648d79f986a96fb5/invenio_app_rdm/redirector/resource.py#L96

However, the `target` can be a callable that returns a `redirect_url` or a tuple `(redirect_url, code)`

We never handle the view arguments. If that is to be done, it must be done in the callable itself.